lib/active_merchant/billing/gateways/hps.rb

```diff
@@ -295,18 +332,19 @@ def commit(action, &request)
         )
       end
 
+      SUCCESSFUL_RESPONSE_CODES = %w(0 00 85)
       def successful?(response)
         (
           (response['GatewayRspCode'] == '0') &&
-          ((response['RspCode'] || '00') == '00' || response['RspCode'] == '85')
```

Comment:
It looks to me like this says "if there's no response code, pretend we got a response code of 00"; this is a little perplexing, but I wasn't convinced that trying to dig into this would be a good use of time.

Reply:
This is indeed a bit perplexing. The approach you took makes enough sense to me. 👍